2200001/문제5.py

```diff
@@ -0,0 +1,32 @@
+# 괄호 짝짓기
+T = 10
+
+for t in range(1, T+1):
+    length = int(input())
+    string = input()
+
+    s_list = []
+    result = 1
+    
+    for i in range(length):
+        if string[i] == '(' or string[i] == '{' or string[i] == '[' or string[i] == '<':
+            s_list.append(string[i])
+
+        if string[i] == ')':
+            if len(s_list) > 0 and s_list.pop() != '(':
+                result = 0
+                break
+        if string[i] == '}':
+            if len(s_list) > 0 and s_list.pop() != '{':
+                result = 0
+                break
+        if string[i] == ']':
+            if len(s_list) > 0 and s_list.pop() != '[':
+                result = 0
+                break
+        if string[i] == '>':
+            if len(s_list) > 0 and s_list.pop() != '<':
+                result = 0
+                break
```

Comment:
깔끔한 풀이네요! 👍
테스트 케이스에서는 없었지만, 입력된 문자열에 닫는 괄호(`)`, `}`, `]`, `>`)가 먼저 나온다면, 리스트에서 마지막 원소에 접근을 할 수 없고, `pop`을 할 수 없어 에러가 발생합니다! 아래와 같은 하나의 테스트 케이스에서 0이 출력되야 합니다.
```text
1
]
```
아래와 같이 수정해주시면 이 문제를 해결할 수 있습니다 😄 
```suggestion
        if string[i] == ')':
            if len(s_list) == 0 or s_list.pop() != '(':
                result = 0
                break
        if string[i] == '}':
            if len(s_list) == 0 or s_list.pop() != '{':
                result = 0
                break
        if string[i] == ']':
            if len(s_list) == 0 or s_list.pop() != '[':
                result = 0
                break
        if string[i] == '>':
            if len(s_list) == 0 or s_list.pop() != '<':
                result = 0
                break
```